src/window_identifier/wayland.rs

```diff
@@ -1,40 +1,79 @@
 use std::fmt;
 
+use wayland_backend::sys::client::Backend;
 use wayland_client::{
-    protocol::{__interfaces::WL_SURFACE_INTERFACE, wl_surface::WlSurface},
+    protocol::{wl_registry, wl_surface::WlSurface},
     Proxy, QueueHandle,
 };
-use wayland_protocols::xdg::foreign::zv2::client::{
-    zxdg_exported_v2::{Event, ZxdgExportedV2},
-    zxdg_exporter_v2::ZxdgExporterV2,
+use wayland_protocols::xdg::foreign::{
+    zv1::client::{
+        zxdg_exported_v1::{self, ZxdgExportedV1},
+        zxdg_exporter_v1::ZxdgExporterV1,
+    },
+    zv2::client::{
+        zxdg_exported_v2::{self, ZxdgExportedV2},
+        zxdg_exporter_v2::ZxdgExporterV2,
+    },
 };
 
 use super::WindowIdentifierType;
 
+// Supported versions.
+const ZXDG_EXPORTER_V1: u32 = 1;
+const ZXDG_EXPORTER_V2: u32 = 1;
+
+#[derive(Debug)]
 pub struct WaylandWindowIdentifier {
-    exported: ZxdgExportedV2,
+    exported_v1: Option<ZxdgExportedV1>,
+    exported_v2: Option<ZxdgExportedV2>,
```

Comment:
We could us an Inner enum where v1 and v2 are variants go make the implementation a bit more idomatic.